airbyte-commons-worker/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java

```diff
@@ -169,6 +173,31 @@ public Process create(final String jobType,
     }
   }
 
+  /**
+   * !! ONLY FOR DEBUGGING, SHOULD NOT BE USED IN PRODUCTION !! If you set the DEBUG_CONTAINER_IMAGE
+   * environment variable, and it matches the image name of a spawned container, this method will add
+   * the necessary params to connect a debugger. For example, to enable this for
+   * `destination-bigquery` start the services locally with: ``` VERSION="dev"
+   * DEBUG_CONTAINER_IMAGE="destination-bigquery" docker compose -f docker-compose.yaml -f
+   * docker-compose.debug.yaml up ``` Additionally you may have to update the image version of your
+   * target image to 'dev' in the UI of your local airbyte platform. See the
+   * `docker-compose.debug.yaml` file for more context.
+   *
+   * @param containerName the name of the container which could be debugged.
+   * @return A list with debugging arguments or an empty list
+   */
+  static List<String> localDebuggingOptions(final String containerName) {
+    final boolean shouldAddDebuggerOptions =
+        Optional.ofNullable(System.getenv("DEBUG_CONTAINER_IMAGE")).filter(StringUtils::isNotEmpty)
```

Comment:
Can you add the environment lookups to https://github.com/airbytehq/airbyte/blob/fca3940d2cf608cfa653a1695196d623bde4876a/airbyte-config/config-models/src/main/java/io/airbyte/config/EnvConfigs.java#L38?

For context, all the config goes through `EnvConfigs`, we're in the process of migrating to micronaut which should also be handling configuration for us: https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/resources/application.yml. I am thinking `airbyte.container.docker.debug-container-java-opts` might make sense for the time being.

Reply:
can do

Reply:
Opting to not implement this change - this is for debugging only and probably should be more broadly integrated into the standard execution process. Could add as a follow on